ElrondNetwork/elrond-go
# add guardian data REST api

## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
This PR adds a rest API for getting the guardian data of an address
  
## Proposed Changes
The guardianData returned on the API will give several useful information such as
 - the active guardian (bech32 if any) and its activation epoch
 - the pending guardian (bech32 if any) and its activation epoch
- frozen state of the account - bool

## Testing procedure
- add guardian on an account and use the API to check the guardian data (one pending guardian)
- wait the expected time for the guardian to become active, then use the API to check the guardian data is as expected (one active guardian)
- add another guardian and use the API to check the guardian data (one active one pending)
- set account in frozen state then check the guardian data (one active guardian, the pending was cleared by freeze account action, and account is in frozen state)
- set another guardian (through an ungarded tx) and check the guardian data (one active, one pending, account frozen)
